pkg/dbs/studydb/db_study-surveys.go

```diff
@@ -110,16 +110,16 @@ func (dbService *StudyDBService) GetSurveyKeysInStudy(instanceID string, studyKe
 
 }
 
-func (dbService *StudyDBService) FindAllCurrentSurveyDefsForStudy(instanceID string, studyKey string, includeUnpublished bool, onlyInfos bool) (surveys []*types.Survey, err error) {
-	surveyKeys, err := dbService.GetSurveyKeysInStudy(instanceID, studyKey, includeUnpublished)
+func (dbService *StudyDBService) FindAllCurrentSurveyDefsForStudy(instanceID string, studyKey string, onlyInfos bool) (surveys []*types.Survey, err error) {
+	surveyKeys, err := dbService.GetSurveyKeysInStudy(instanceID, studyKey, false)
 	if err != nil {
 		return surveys, err
 	}
 	for _, key := range surveyKeys {
 		survey, err := dbService.FindCurrentSurveyDef(instanceID, studyKey, key, onlyInfos)
 		if err != nil {
 			logger.Error.Println(err)
-			continue
+			return surveys, err
```

Comment:
maybe we should return nil, err and not a partial list of surveys in case of errors

Reply:
ok